test/smoke/go/operator.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"time"
+	
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/api/core/v1"
+	coreapps "k8s.io/api/apps/v1"
+	rbac "k8s.io/api/rbac/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+	"github.com/kiegroup/kogito-cloud-operator/version"
+)
+
+const (
+	DEFAULT_OPERATOR_IMAGE="quay.io/kiegroup/kogito-cloud-operator"
+	
+	OPERATOR_DEPLOY_DEFAULT_URI = "https://raw.githubusercontent.com/kiegroup/kogito-cloud-operator/master/deploy/"
+	OPERATOR_NAME = "kogito-cloud-operator"
+	
+	KOGITO_CRD_GROUP_NAME = "app.kiegroup.org";
+    KOGITOAPP_CRD_NAME = "kogitoapps";
+    KOGITOINFRA_CRD_NAME = "kogitoinfras";
+    KOGITODATAINDICES_CRD_NAME = "kogitodataindices"
+)
+
+var (
+	// TODO should take the version as argument
+	operatorImageNameTag = fmt.Sprintf("%s:%s", DEFAULT_OPERATOR_IMAGE, version.Version)
+)
+
+func DeployOperatorFromYaml(namespace string) error{
+	// Create crds files if needed
+	if err := deployCrdIfNeeded(KOGITOAPP_CRD_NAME); err != nil { return err }
+	if err := deployCrdIfNeeded(KOGITOINFRA_CRD_NAME); err != nil { return err }
+	if err := deployCrdIfNeeded(KOGITODATAINDICES_CRD_NAME); err != nil { return err }
+	
+	var deployUri = getOperatorDeployUri()
+	fmt.Println("Deploy Operator from yaml files in " + deployUri)
+	
+	loadResource(namespace, deployUri + "service_account.yaml",  &corev1.ServiceAccount{}, nil)
+	loadResource(namespace, deployUri + "role.yaml",  &rbac.Role{}, nil)
+	loadResource(namespace, deployUri + "role_binding.yaml",  &rbac.RoleBinding{}, nil)
+	loadResource(namespace, deployUri + "operator.yaml",  &coreapps.Deployment{}, func(object interface{}) {
+		object.(*coreapps.Deployment).Spec.Template.Spec.Containers[0].Image = operatorImageNameTag
+	})
+	
+	return nil
+}
+
+func IsOperatorRunning(namespace string) (bool,error) {
+	if operatorDeployment,err := GetOperatorDeployment(namespace); err != nil {
+		return false, err
+	} else if operatorDeployment == nil {
+		return false, nil
+	} else if operatorDeployment.Status.AvailableReplicas == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+func GetOperatorDeployment(namespace string) (*coreapps.Deployment, error) {
```

Comment:
We have this:
https://github.com/kiegroup/kogito-cloud-operator/blob/1e89c86ddc2e28728cffa3b78f899d2c6e110170/cmd/kogito/command/shared/resource_checks.go#L70

Reply:
Reused it. Was just not a big fan reusing some logic from cli or operator controller.
I reuse client which is more or less generic and could be reused in other projects
I do need the apis and a change in it, I will see it really fast.
Reuse the logic can be risky IMO. But for now I reuse it.

Reply:
or we would need to create generic package with those methods.

Reply:
We have to move this function to a shared package. I suggest `pkg/infrastructure`.